api/src/test/java/com/epam/pipeline/controller/cluster/ClusterControllerTest.java

```diff
@@ -0,0 +1,309 @@
+
+        final MvcResult mvcResult = performRequest(get(String.format(LOAD_NODE_URL, NAME)));
+
+        Mockito.verify(mockClusterApiService).getNode(NAME);
+        assertResponse(mvcResult, nodeInstance, NodeCreatorUtils.NODE_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailLoadNodeFilteredForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(post(LOAD_NODES_URL, NAME));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadNodeFiltered() throws Exception {
+        final FilterPodsRequest filterPodsRequest = NodeCreatorUtils.getDefaultFilterPodsRequest();
+        final String content = getObjectMapper().writeValueAsString(filterPodsRequest);
+        Mockito.doReturn(nodeInstance).when(mockClusterApiService)
+                .getNode(Mockito.eq(NAME), Mockito.refEq(filterPodsRequest));
+
+        final MvcResult mvcResult = performRequest(post(String.format(LOAD_NODE_URL, NAME)).content(content));
+
+        Mockito.verify(mockClusterApiService)
+                .getNode(Mockito.eq(NAME), Mockito.refEq(filterPodsRequest));
+        assertResponse(mvcResult, nodeInstance, NodeCreatorUtils.NODE_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailTerminateNodeForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(delete(NODE_NAME_URL, NAME));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldTerminateNode() throws Exception {
+        Mockito.doReturn(nodeInstance).when(mockClusterApiService).terminateNode(NAME);
+
+        final MvcResult mvcResult = performRequest(delete(String.format(NODE_NAME_URL, NAME)));
+
+        Mockito.verify(mockClusterApiService).terminateNode(NAME);
+        assertResponse(mvcResult, nodeInstance, NodeCreatorUtils.NODE_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailLoadAllInstanceTypesForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(LOAD_INSTANCE_TYPES_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAllowedInstanceTypes() throws Exception {
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("regionId", "1");
+        params.add("toolInstances", "false");
+        params.add("spot", "true");
+        Mockito.doReturn(instanceTypes).when(mockClusterApiService).getAllowedInstanceTypes(ID, true);
+
+        final MvcResult mvcResult = performRequest(get(LOAD_INSTANCE_TYPES_URL).params(params));
+
+        Mockito.verify(mockClusterApiService).getAllowedInstanceTypes(ID, true);
+        assertResponse(mvcResult, instanceTypes, NodeCreatorUtils.INSTANCE_TYPE_LIST_TYPE);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAllowedToolInstanceTypes() throws Exception {
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("regionId", "1");
+        params.add("toolInstances", "true");
```

Comment:
Could you please also add the similar test case which covers `"toolInstances": false`?